director/adjustable_client.go

```diff
@@ -41,6 +41,7 @@ func (c AdjustableClient) Do(req *http.Request) (*http.Response, error) {
 	}
 
 	if c.adjustment.NeedsReadjustment(resp) {
+		resp.Body.Close()
```

Comment:
@aeijdenberg Can you help me understand why this fixes what you're seeing? I think this `Close` close the request body file descriptor. Which would mean subsequent requests will most definitely fail if it has a body and is trying to reuse this? I'd expect a `file already closed` type of error on retry.

Reply:
@dpb587-pivotal - thanks for looking at this PR. Once we get inside of this `if` statement, the value `resp` is no longer used (I agree this function as written is quite confusing).

i.e. reaching this point, the function will either return on line 47, or line 52. In either case, `resp` is not accessed, and thus is never closed.

Reply:
Ah, I read too quickly. This is the response body, not the request body. Thanks.